packages/hdwallet-keepkey/src/keepkey.ts

```diff
@@ -58,6 +60,78 @@ export function isKeepKey(wallet: any): wallet is KeepKeyHDWallet {
   return typeof wallet === 'object' && wallet._isKeepKey !== undefined
 }
 
+function describeETHPath (path: BIP32Path) {
+  if (path.length != 5)
+    return undefined
+
+  if (path[0] != 0x80000000 + 44)
+    return undefined
+
+  if (path[1] != 0x80000000 + slip44ByCoin('Ethereum'))
+    return undefined
+
+  if ((path[2] & 0x80000000) !== 0x80000000)
+    return undefined
+
+  if (path[3] != 0)
+    return undefined
+
+  if (path[4] != 0)
+    return undefined
+
+  let index = path[2] & 0x7fffffff
+  return `Ethereum Account #${index}`
+}
+
+function describeUTXOPath (path: BIP32Path, coin: Coin, scriptType: BTCInputScriptType) {
+  if (!this.btcSupportsCoin(coin))
+    return undefined
+
+  if (!this.btcSupportsScriptType(coin, scriptType))
+    return undefined
+
+  if (path.length !== 3 && path.length !== 5)
+    return undefined
+
+  if ((path[0] & 0x80000000) !== 0x80000000)
+    return undefined
+
+  let purpose = path[0] & 0x7fffffff
+
+  if (![44, 49, 84].includes(purpose))
+    return undefined
+
+  if (purpose === 44 && scriptType !== BTCInputScriptType.SpendAddress)
+    return undefined
+
+  if (purpose === 49 && scriptType !== BTCInputScriptType.SpendP2SHWitness)
+    return undefined
+
+  if (purpose === 84 && scriptType !== BTCInputScriptType.SpendWitness)
+    return undefined
+
+  if (path[1] !== 0x80000000 + slip44ByCoin(coin))
+    return undefined
+
+  let wholeAccount = path.length === 3
+
+  let script = {
+    [BTCInputScriptType.SpendAddress]: '',
+    [BTCInputScriptType.SpendP2SHWitness]: 'Segwit ',
+    [BTCInputScriptType.SpendWitness]: 'Segwit Native '
+  }[scriptType]
+
+  let index = path[2] & 0x7fffffff
+
+  if (wholeAccount) {
+    return `${coin} ${script}Account #${index}`
+  } else {
+    let change = path[3] == 1 ? 'Change ' : ''
+    let address = path[4]
+    return `${coin} ${script}Account #${index}, ${change}Address #${address}`
```

Comment:
@ascension I'm considering returning an object from these, something like:

```
return {
  verbose: `${coin} ${script}Account #${index}, ${change}Address #${address}`,
  accountIdx: index,
  addressIdx: address,
  change: path[3] === 1
}
```

Reply:
Can we add `coin` as a property.

Reply:
so just the value that got passed in? seems not too useful